src/opnsense/scripts/shell/setaddr.php

```diff
@@ -259,26 +269,26 @@ function add_gateway_to_config($interface, $gatewayip, $inet_type, $is_in_subnet
         echo "\n";
     }
 
-    if ($new_name == '') {
+    if (empty($new_name)) {
         $new_name = next_unused_gateway_name($interface);
     }
 
-    $item = array(
+    $item = [
+        'disabled' => 0,
         'descr' => sprintf('Interface %s Gateway', strtoupper($interface)),
-        'defaultgw' => $is_default,
+        'defaultgw' => $is_default ? 1 : 0,
         'ipprotocol' => $inet_type,
         'interface' => $interface,
         'gateway' => $gatewayip,
         'monitor_disable' => 1,
         'name' => $new_name,
-        'interval' => true,
         'weight' => 1,
-    );
-    if (!$is_in_subnet) {
-        $item['fargw'] = 1;
-    }
+        'fargw' => !$is_in_subnet ? 1 : 0
+    ];
 
-    $a_gateways[] = $item;
+    $gw->createOrUpdateGateway($item, $uuid);
+    /* Serialize back to global config */
+    $config = OPNsense\Core\Config::getInstance()->toArray(listtags());
```

Comment:
ugh :)

Reply:
doesn't this attempt to do "write_config()" later on?

Reply:
yes, but it isn't the same instance, result is that the latter `write_config()` will overwrite the changes made here

Reply:
I know, maybe the comment needs to be improved. it explains what but not why

Reply:
sure, and no, it isn't pretty

Reply:
to be fair the problem is real, but before write_config() would actually serialize all of the changes at once. Maybe the code doing the config sync could be moved closer together or the $config is rewritten in memory but not on the disk for the gateway flush?

Reply:
sorry maybe it's doing that. still on my first coffee